api/user/views.py

```diff
@@ -34,6 +34,32 @@ def delete(request, *args, **kwargs):
             user_id = str(request.user.id)
             use_case.execute(user_id)
             return Response({"message": "User deleted successfully"}, status=status.HTTP_204_NO_CONTENT)
+        except Exception as e:
+            status_response, detail = get_error_status_code_from_exception(e)
+            return Response(data={"error": detail}, status=status_response)
+
+    @staticmethod
+    def put(request, *args, **kwargs):
+        repository = UserRepositoryImplements()
+        use_case = UpdateUserUseCase(user_repository=repository)
+        try:
+            user_id = request.user.id
+            user_data = request.data
+            updated_user = use_case.execute(user_id, user_data)
+            return Response(updated_user, status=200)
+        except Exception as e:
+            status_response, detail = get_error_status_code_from_exception(e)
+            return Response(data=detail, status=status_response)
+
+class UserUpdateByIdAPIView(APIView):
+    @staticmethod
+    def put(request, user_id, *args, **kwargs):
+        repository = UserRepositoryImplements()
+        use_case = UpdateUserUseCase(user_repository=repository)
+        try:
+            user_data = request.data
+            updated_user = use_case.execute(user_id, user_data)
```

Comment:
you should explicitly tell which props do you need from data, otherwise you could make it possible to update data not meant to be able to, for example, created_by, or other data that is not supposed to be able to be updatable, thats a security risk, and a bad practice, take it in consideration
```suggestion
            user_data = request.data
            updated_user = use_case.execute(user_id, UserUpdateDto(
              email=request.data.get("email"),
              first_name=request.data.get("first_name"),
              # ...
            ))
```